bumble/gatt_client.py

```diff
@@ -942,6 +948,11 @@ def on_att_handle_value_notification(self, notification):
         )
         if not subscribers:
             logger.warning('!!! received notification with no subscriber')
+
+        self.cached_values[notification.attribute_handle] = (
```

Comment:
The same caching logic appears 3 times. This could be factored out in a `cache_value(handle, value)` method.